src/Datadog.Trace/Tracer.cs

```diff
@@ -361,6 +366,130 @@ private static IStatsd CreateDogStatsdClient(TracerSettings settings, string ser
+
+                                if (!string.IsNullOrWhiteSpace(path))
+                                {
+                                    if (ProgramIsRunning(path))
+                                    {
+                                        DatadogLogging.RegisterStartupLog(log => log.Debug("{0} at {1} is already running.", processPathVariable, path));
+                                        continue;
+                                    }
+
+                                    var startInfo = new ProcessStartInfo { FileName = path };
+
+                                    if (!string.IsNullOrWhiteSpace(args))
+                                    {
+                                        startInfo.Arguments = args;
+                                    }
+
+                                    DatadogLogging.RegisterStartupLog(log => log.Debug("Starting {0} at {1}.", processPathVariable, path));
+
+                                    var process = Process.Start(startInfo);
+
+                                    Thread.Sleep(150);
+
+                                    if (process == null || process.HasExited)
+                                    {
+                                        DatadogLogging.RegisterStartupLog(log => log.Error("{0} at {1} has failed to start.", processPathVariable, path));
+                                        sequentialFailures++;
+                                    }
+                                    else
+                                    {
+                                        DatadogLogging.RegisterStartupLog(log => log.Debug("Successfully started {0} at {1}.", processPathVariable, path));
+                                        sequentialFailures = 0;
+                                    }
+                                }
+                                else
+                                {
+                                    DatadogLogging.RegisterStartupLog(log => log.Debug("There is no path configured for {0}.", processPathVariable));
+                                    return;
+                                }
+                            }
+                            catch (Exception ex)
+                            {
+                                DatadogLogging.RegisterStartupLog(log => log.Error(ex, "Exception when trying to start an instance of {0} at {1}.", processPathVariable, path));
+                                sequentialFailures++;
+                            }
+                            finally
+                            {
+                                // Make sure to delay every loop
+                                Thread.Sleep(20_000);
+                            }
+
+                            if (sequentialFailures >= circuitBreakerMax)
+                            {
+                                DatadogLogging.RegisterStartupLog(log => log.Error("Circuit breaker triggered for {0} at {1}.", processPathVariable, path));
```

Comment:
Can you add into the log message that the sequential failures exceeded some limit? Hypothetically there could be multiple reasons for circuit breaker being triggered